### PR TITLE
Remove the deprecating armeabi ABI from build

### PR DESCRIPTION
**Pull request by ggfan**

### CODE REVIEW

**Comment by ggfan** [comment]
Purpose:
- remove armeabi
- use default for ABI when possible for samples ( for r16, the default is x86, x86-64, armeabi-v7a, arm64-v8a)
- For ndk-build + gradle, remove explicit clang usage since it is default for r16

will add another branch to deal with CMake's usage for default stl and compiler after fully configuring one or two samples.

@DanAlbert  @jomof please take a look

File: hello-jni/app/build.gradle [hunk 1]
```diff
@@ -58,16 +52,10 @@ android {
...
-            dimension 'cpuArch'
-            ndk {
-                abiFilters 'mips', 'mips64'
-            }
-        }
         universal {
             dimension 'cpuArch'
             ndk {
-                abiFilters 'mips', 'mips64', 'x86', 'x86_64'
+                abiFilters 'x86', 'x86_64'
```
Comment (DanAlbert) [inline hello-jni/app/build.gradle]: What's this? How is it universal if it only targets half of the architectures?

File: other-builds/ndkbuild/hello-neon/app/build.gradle [hunk 2]
```diff
@@ -37,10 +37,6 @@ android {
             dimension "abi"
             ndk.abiFilters 'armeabi-v7a'
         }
-        x86 {
```
Comment (DanAlbert) [inline other-builds/ndkbuild/hello-neon/app/build.gradle]: I don't speak gradle, but isn't this removing support for x86* from this sample?

File: hello-jni/app/build.gradle (hunk 1 above)
Comment (ggfan) [reply hello-jni/app/build.gradle]: this is just an example to show that application could pick up any kinds of combination of the ABI into flavor build.

File: other-builds/ndkbuild/hello-neon/app/build.gradle (hunk 2 above)
Comment (ggfan) [reply other-builds/ndkbuild/hello-neon/app/build.gradle]: this is a battle: like not to build for x86 as the performance number on x86 is slower than the C version of the sample;  it makes great senses for armeabi-v7a though.

Comment (DanAlbert) [reply other-builds/ndkbuild/hello-neon/app/build.gradle]: So? It's a sample. If the NEON emulation performance isn't even good enough for a sample we should just remove it entirely.

File: hello-jni/app/build.gradle (hunk 1 above)
Comment (DanAlbert) [reply hello-jni/app/build.gradle]: I still don't understand how this is in any way "universal", but it's not related to the rest of the patch, so whatever. Should probably add some comments explaining why this is the way it is (I sure don't understand it and as far as gradle knowledge goes, I am the target user of the samples), but that can be a different patch.

File: other-builds/ndkbuild/hello-neon/app/build.gradle (hunk 2 above)
Comment (ggfan) [reply other-builds/ndkbuild/hello-neon/app/build.gradle]: probably the best action now is: keep the x86 build as-is ( the purpose of the PR is to get rid of the warning of armeabi in r16).
created a bug to look at it deeply:  https://github.com/googlesamples/android-ndk/issues/472

File: hello-jni/app/build.gradle (hunk 1 above)
Comment (ggfan) [reply hello-jni/app/build.gradle]: yeah, will look at that with next patch for the compiler and stl default handling.

**Pull request merged**